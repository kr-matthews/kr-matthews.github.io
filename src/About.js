import {Link} from 'react-router-dom';

function About(props) {
  return (
    <div className="page narrow-page">
      <h1>About</h1>
      <p>
      I'm Kevin Matthews, a mathematician by training.
      Generally, I'm interested in most kinds of problem solving
      related to mathematics and programming.
      I like systems design --
      how software applications are structured,
      how cities function,
      how organizations setup their workflows,
      and such.
      </p>
      <p>
      I have a Bachelor of Mathematics from the University of Waterloo,
      where I studied Pure Math, Applied Math, Combinatorics & Optimization,
      and Computational Math.
      I also have a Master of Arts from the University of California, Los Angeles,
      where I studied Mathematical Logic.
      More recently, I've been working behind-the-scenes
      with financial planning/advising workflows and processes.
      </p>
      <p>
      I know how to solve a Rubik's cube (blindfolded),
      and have <a href="https://www.worldcubeassociation.org/persons/2010MATT02">
      competed</a> in about 100 competitions across North America,
      a few of which I helped organize.
      Most of my early programming <Link to="/projects">projects</Link> were
      cube-based tools, designed to help with learning and/or practising the cube.
      </p>
      <p>
      I'm currently in BC, Canada. I can be contacted at my email,
      which is 'kvn' then 'r' followed by my last name,
      at Google's mail platform.
      </p>
    </div>
  )
}

export default About;
