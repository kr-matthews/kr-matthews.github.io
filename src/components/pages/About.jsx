import { Fragment } from "react";
import Link from "../common/Link";
import { NarrowContent } from "../common/Page";
import Tabs from "../common/Tabs";

const tabData = [
  { name: "Education & Work", content: <EducationAndWork /> },
  { name: "Projects & Hobbies", content: <ProjectsAndHobbies /> },
  { name: "Contact", content: <Contact /> },
];

export default function About() {
  return (
    <NarrowContent>
      <h1>About</h1>

      <p style={{ marginBottom: "2em" }}>
        I'm Kevin Matthews, a mathematician by training and a self-taught
        programmer. Generally, I'm interested in most kinds of problem solving
        related to mathematics and programming. Among other things, I like
        systems design -- how software applications are structured, how cities
        function, how organizations setup their workflows, and such.
      </p>

      <Tabs data={tabData} />
    </NarrowContent>
  );
}

function EducationAndWork() {
  return (
    <Fragment>
      <p>
        I have a Bachelor of Mathematics from the{" "}
        <Link href="https://uwaterloo.ca/math/" isExternal>
          University of Waterloo
        </Link>
        , where I studied Pure Math, Applied Math, Combinatorics & Optimization,
        and Computational Math. I also have a Master of Arts from the{" "}
        <Link href="https://ww3.math.ucla.edu/" isExternal>
          University of California, Los Angeles
        </Link>
        , where I studied Mathematical Logic.
      </p>
      <p>
        In 2022 I joined{" "}
        <Link href="https://ensemble.com/" isExternal>
          Ensemble Systems
        </Link>{" "}
        as a Software Developer. Currently I'm working on an Android project,
        and before that I was leading the front-end development of a React web
        project .
      </p>
      <p>
        Before that, I worked as a Software Engineer in Android development at{" "}
        <Link href="https://rivian.com/" isExternal>
          Rivian
        </Link>
        , an electric vehicle company. My main focus was on setup/pairing flows
        for devices via Bluetooth low energy, but I also assisted other teams as
        needed, and was part of several app-wide initiatives to improve
        reusability and consistency.
      </p>
      <p>
        Beyond that, I have some experience working in financial
        management/planning, and was a teaching assistant (or instructional
        support assistant) for various computer science and math courses at UW
        and UCLA.
      </p>
    </Fragment>
  );
}

function ProjectsAndHobbies() {
  return (
    <Fragment>
      <p>
        Since 2021, I've been consistently working on numerous personal{" "}
        <Link to="/projects">projects</Link> to improve my skills while learning
        new languages and techniques. Most are simple games, but there are also
        some more useful things, such as an integration with the Strava API, a
        bike route map of Metro Vancouver, and a reusable footer component for
        use across my various projects. I also have ideas for other types of
        projects in the future.
      </p>
      <p>
        I've been solving <Link to="/cubing">Rubik's Cubes</Link> for over a
        decade, and have helped organize and run numerous competitions as a
        volunteer, achieving a few Canadian records along the way. Most of my
        pre-2021 programming <Link to="/projects">projects</Link> were
        cube-based tools, designed to help with learning how to solve the cube
        blindfolded and to help practice what you know. I'm also on the World
        Cube Association's{" "}
        <Link href="https://github.com/thewca" isExternal>
          software team
        </Link>{" "}
        as a volunteer, where my main focuses include assisting on transitioning
        our website's front-end from Rails to React, and building out a new
        separate registration system.
      </p>
      <p>
        I also like playing <Link to="/vancouver">GeoGuessr</Link> restricted to
        Vancouver (or subsets of Metro Vancouver) with 10s NMPZ (no moving,
        panning, or zooming).
      </p>
      <p>
        Other hobbies include reading, cycling, running, soccer, table tennis,
        tetris attack (this isn't tetris), and more.
      </p>
    </Fragment>
  );
}

function Contact() {
  return (
    <Fragment>
      <p>
        I'm currently in <Link to="/vancouver">Vancouver</Link>, BC, Canada. I
        can be contacted at my email, which is 'kvn' then 'r' followed by my
        last name, at Google's mail platform.
      </p>
    </Fragment>
  );
}
