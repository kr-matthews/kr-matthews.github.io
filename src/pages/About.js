import { Link } from "react-router-dom";

export default function About(props) {
  return (
    <div className="page narrow-page">
      <h1>About</h1>

      <h3>Interests</h3>
      <p>
        I'm Kevin Matthews, a mathematician by training and a self-taught
        programmer. Generally, I'm interested in most kinds of problem solving
        related to mathematics and programming. Among other things, I like
        systems design -- how software applications are structured, how cities
        function, how organizations setup their workflows, and such.
      </p>

      <h3>Education and Work</h3>
      <p>
        I have a Bachelor of Mathematics from the{" "}
        <a
          href="https://uwaterloo.ca/math/"
          target="_blank"
          rel="noopener noreferrer"
        >
          University of Waterloo
        </a>
        , where I studied Pure Math, Applied Math, Combinatorics & Optimization,
        and Computational Math. I also have a Master of Arts from the{" "}
        <a
          href="https://ww3.math.ucla.edu/"
          target="_blank"
          rel="noopener noreferrer"
        >
          University of California, Los Angeles
        </a>
        , where I studied Mathematical Logic.
      </p>
      <p>
        I worked most recently as a Software Engineer in Android development at{" "}
        <a href="https://rivian.com/" target="_blank" rel="noopener noreferrer">
          Rivian
        </a>
        , an electric vehicle company, where I learnt a lot of new techniques
        and skills. My main focus was on setup/pairing flows for devices via
        Bluetooth low energy, but I also assisted other teams as needed, and was
        part of several app-wide initiatives to improve reusability and
        consistency. Before that, I worked in Financial Technology for a couple
        of years, focusing on behind-the-scenes financial planning workflows and
        processes.
      </p>

      <h3>Projects and Hobbies</h3>
      <p>
        For the past year or so, I've been consistently working on numerous
        personal <Link to="/projects">projects</Link> to improve my skills while
        learning new languages and techniques. Most are simple games, but I have
        ideas for other types of projects in the future.
      </p>
      <p>
        I've been solving <Link to="/cubing">Rubik's Cubes</Link> for over a
        decade, and have helped organize and run numerous competitions as a
        volunteer, achieving a few Canadian records along the way. Most of my
        early programming <Link to="/projects">projects</Link> were cube-based
        tools, designed to help with learning how to solve the cube blindfolded
        and to help practice what you know. I'm also on the World Cube
        Association's{" "}
        <a
          href="https://github.com/thewca"
          target="_blank"
          rel="noopener noreferrer"
        >
          software team
        </a>{" "}
        as a volunteer, where my main focus is assisting on transitioning our
        website's front-end from Rails to React.
      </p>

      <h3>Contact</h3>
      <p>
        I'm currently in <Link to="/vancouver">Vancouver</Link>, BC, Canada. I
        can be contacted at my email, which is 'kvn' then 'r' followed by my
        last name, at Google's mail platform.
      </p>
    </div>
  );
}
