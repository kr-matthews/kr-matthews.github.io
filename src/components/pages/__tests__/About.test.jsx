import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import About from "../About";
import { BrowserRouter } from "react-router-dom";

describe("components/pages/About", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<About />, {
      wrapper: ({ children }) => <BrowserRouter>{children}</BrowserRouter>,
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
