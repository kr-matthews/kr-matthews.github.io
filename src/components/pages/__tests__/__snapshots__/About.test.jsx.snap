// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/About matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-rzly3e"
  >
    <div
      class="css-jydvbh"
    >
      <h1>
        About
      </h1>
      <p
        style="margin-bottom: 2em;"
      >
        I'm Kevin Matthews, a mathematician by training and a self-taught programmer. Generally, I'm interested in most kinds of problem solving related to mathematics and programming. Among other things, I like systems design -- how software applications are structured, how cities function, how organizations setup their workflows, and such.
      </p>
      <div
        class="chakra-tabs css-0"
      >
        <div
          aria-orientation="horizontal"
          class="chakra-tabs__tablist css-k008qs"
          role="tablist"
        >
          <button
            aria-controls="tabs-:r0:--tabpanel-0"
            aria-selected="true"
            class="chakra-tabs__tab css-5wl5ud"
            data-index="0"
            id="tabs-:r0:--tab-0"
            role="tab"
            tabindex="0"
            type="button"
          >
            Education & Work
          </button>
          <button
            aria-controls="tabs-:r0:--tabpanel-1"
            aria-selected="false"
            class="chakra-tabs__tab css-5wl5ud"
            data-index="1"
            id="tabs-:r0:--tab-1"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Projects & Hobbies
          </button>
          <button
            aria-controls="tabs-:r0:--tabpanel-2"
            aria-selected="false"
            class="chakra-tabs__tab css-5wl5ud"
            data-index="2"
            id="tabs-:r0:--tab-2"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Contact
          </button>
        </div>
        <div
          class="chakra-tabs__tab-panels css-8atqhb"
        >
          <div
            aria-labelledby="tabs-:r0:--tab-0"
            class="chakra-tabs__tab-panel css-sjt5nk"
            id="tabs-:r0:--tabpanel-0"
            role="tabpanel"
            tabindex="0"
          >
            <p>
              I have a Bachelor of Mathematics from the 
              <a
                class="chakra-link css-1108m31"
                href="https://uwaterloo.ca/math/"
                rel="noopener"
                target="_blank"
                to=""
              >
                University of Waterloo
              </a>
              , where I studied Pure Math, Applied Math, Combinatorics & Optimization, and Computational Math. I also have a Master of Arts from the 
              <a
                class="chakra-link css-1108m31"
                href="https://ww3.math.ucla.edu/"
                rel="noopener"
                target="_blank"
                to=""
              >
                University of California, Los Angeles
              </a>
              , where I studied Mathematical Logic.
            </p>
            <p>
              In 2022 I joined 
              <a
                class="chakra-link css-1108m31"
                href="https://ensemble.com/"
                rel="noopener"
                target="_blank"
                to=""
              >
                Ensemble Systems
              </a>
               as a Software Developer. Currently I'm working with smart TVs using Preact. Previously, I was on an Android project using Kotlin, and before that I was leading the front-end development of a React web project.
            </p>
            <p>
              Before that, I worked as a Software Engineer in Android development at 
              <a
                class="chakra-link css-1108m31"
                href="https://rivian.com/"
                rel="noopener"
                target="_blank"
                to=""
              >
                Rivian
              </a>
              , an electric vehicle company. My main focus was on setup/pairing flows for devices via Bluetooth low energy, but I also assisted other teams as needed, and was part of several app-wide initiatives to improve reusability and consistency.
            </p>
            <p>
              Beyond that, I have some experience working in financial management/planning, and was a teaching assistant (or instructional support assistant) for various computer science and math courses at UW and UCLA.
            </p>
          </div>
          <div
            aria-labelledby="tabs-:r0:--tab-1"
            class="chakra-tabs__tab-panel css-sjt5nk"
            hidden=""
            id="tabs-:r0:--tabpanel-1"
            role="tabpanel"
            tabindex="0"
          >
            <p>
              Since 2021, I've been consistently working on numerous personal 
              <a
                class="chakra-link css-1108m31"
                href="/projects"
              >
                projects
              </a>
               to improve my skills while learning new languages and techniques. Most are simple games, but there are also some more useful things, such as an integration with the Strava API, a bike route map of Metro Vancouver, and a reusable footer component for use across my various projects. I also have ideas for other types of projects in the future.
            </p>
            <p>
              I've been solving 
              <a
                class="chakra-link css-1108m31"
                href="/cubing"
              >
                Rubik's Cubes
              </a>
               for over a decade, and have helped organize and run numerous competitions as a volunteer, achieving a few Canadian records along the way. Most of my pre-2021 programming 
              <a
                class="chakra-link css-1108m31"
                href="/projects"
              >
                projects
              </a>
               were cube-based tools, designed to help with learning how to solve the cube blindfolded and to help practice what you know. I'm also on the World Cube Association's 
              <a
                class="chakra-link css-1108m31"
                href="https://github.com/thewca"
                rel="noopener"
                target="_blank"
                to=""
              >
                software team
              </a>
               as a volunteer, where my main focuses include assisting on transitioning our website's front-end from Rails to React, and building out a new separate registration system.
            </p>
            <p>
              I also like playing 
              <a
                class="chakra-link css-1108m31"
                href="/vancouver"
              >
                GeoGuessr
              </a>
               restricted to Vancouver (or subsets of Metro Vancouver) with 10s NMPZ (no moving, panning, or zooming).
            </p>
            <p>
              Other hobbies include reading, cycling, running, soccer, table tennis, tetris attack (this isn't tetris), and more.
            </p>
          </div>
          <div
            aria-labelledby="tabs-:r0:--tab-2"
            class="chakra-tabs__tab-panel css-sjt5nk"
            hidden=""
            id="tabs-:r0:--tabpanel-2"
            role="tabpanel"
            tabindex="0"
          >
            <p>
              I'm currently in 
              <a
                class="chakra-link css-1108m31"
                href="/vancouver"
              >
                Vancouver
              </a>
              , BC, Canada. I can be contacted at my email, which is 'kvn' then 'r' followed by my last name, at Google's mail platform.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
